File: src/lib/scala/Slate.Core/src/main/scala/slate/core/common/mods/ModuleContext.scala
/**
  * <slate_header>
  * author: Kishore Reddy
  * url: https://github.com/kishorereddy/scala-slate
  * copyright: 2016 Kishore Reddy
  * license: https://github.com/kishorereddy/scala-slate/blob/master/LICENSE.md
  * desc: a scala micro-framework
  * usage: Please refer to license on github for more info.
  * </slate_header>
  */
package slate.core.common.mods

import slate.entities.core.Entities
import slate.entities.models.EntitySetupService

class ModuleContext {

  /**
    * entities for registration/creation
    */
  var entities:Entities = null


  /**
    * referece to service for performing checks/operations on module and status.
    */
  var modService:ModService = null


  /**
    * service to add / manage entities models.
    */
  var modelService:EntitySetupService = null


  var apis:AnyRef = null

}

File: src/support/integration/src/main/kotlin/kiit/integration/mods/ModuleContext.kt
/**
 * <slate_header>
 * url: www.slatekit.com
 * git: www.github.com/code-helix/slatekit
 * org: www.codehelix.co
 * author: Kishore Reddy
 * copyright: 2016 CodeHelix Solutions Inc.
 * license: refer to website and/or github
 * about: A tool-kit, utility library and server-backend
 * mantra: Simplicity above all else
 * </slate_header>
 */

package kiit.integration.mods

import kiit.migrations.MigrationService

data class ModuleContext(

    /**
     * referece to service for performing checks/operations on module and status.
     */
    val service: ModService,

    /**
     * service to add / manage entities models.
     */
    val setup: MigrationService
)